rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml

```diff
@@ -11,3 +11,59 @@ sinks:
     patterns:
       - "ngx-auth-firebaseui|@asithade\\/ngx-auth-firebaseui"
     tags:
+
+  - id: ThirdParties.SDK.Firebaseapp.React-firebase-hooks
+    name: Firebaseapp React-firebase-hooks
+    domains:
+      - "react-firebase-hooks.firebaseapp.com"
+    patterns:
+      - "react-firebase-hooks|@use-firebase/app"
+    tags:
+
+  - id: ThirdParties.SDK.Firebaseapp.Firebase-admin
+    name: Firebaseapp Firebase-admin
+    domains:
+      - "firebase.googleapis.com"
+    patterns:
+      - "firebase-admin|@firebase/admin"
+    tags:
+
+  - id: Firebase.SDK.Authentication
+    name: Firebase Authentication
+    domains:
+      - "firebase.com", "google.com", "facebook.com", "apple.com"
+    patterns:
+      # There is no specific regex pattern for Firebase Authentication, as the SDK allows users to sign in using multiple providers, each with their own pattern.
```

Comment:
@hiaashna Can you share the examples here? If we don't have the pattern the rule will not be of much use

Reply:
I agree that without a specific pattern, the rule provided for Firebase Authentication may not be as useful in detecting personal data. The Firebase Authentication SDK allows users to sign in with multiple providers, such as Google, Facebook and each provider may have a different format for the personal data they collect.

few eg:
- Google: pattern: "(?i)("id_token":")(.*)(","access_token")"
- Facebook: pattern: "(?i)("access_token":")(.*)(","token_type")"